### PR TITLE
Adding Support for OpenSSL 1.1.1 (SOFTWARE-4150)

### PR DESCRIPTION
**Pull request by efajardo**

### CODE REVIEW

File: src/GlobusSupport.cc [hunk 1]
```diff
@@ -315,7 +315,11 @@ class CertStore {
                                   globus_gsi_callback_create_proxy_callback);
     X509_STORE_set_depth(m_cert_store, GLOBUS_GSI_CALLBACK_VERIFY_DEPTH);
     X509_STORE_set_flags(m_cert_store, X509_V_FLAG_ALLOW_PROXY_CERTS);
-    m_cert_store->check_issued = globus_gsi_callback_check_issued;
+    #ifdef _OPENSSL111
```
Comment (edquist) [inline src/GlobusSupport.cc:318]:
```suggestion
   #ifdef _OPENSSL111
```

whitespace

File: src/GlobusSupport.cc [hunk 2]
```diff
@@ -315,7 +315,11 @@ class CertStore {
...
+        X509_STORE_set_check_issued(m_cert_store, globus_gsi_callback_check_issued);
+   #else
+	m_cert_store->check_issued = globus_gsi_callback_check_issued;
+   #endif
```
Comment (edquist) [inline src/GlobusSupport.cc:322]:
```suggestion
    #ifdef _OPENSSL111
        X509_STORE_set_check_issued(m_cert_store, globus_gsi_callback_check_issued);
    #else
	m_cert_store->check_issued = globus_gsi_callback_check_issued;
    #endif
```

File: rpm/xrootd-lcmaps.spec [hunk 3]
```diff
@@ -51,8 +51,12 @@ Requires: xrootd-server <  1:%{xrootd_next}.0-0
 scl enable devtoolset-2 '
 %endif
 
-#cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=RelWithDebInfo .
+%if 0%{?rhel} >= 8
+%cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DOPENSSL111=1 .
```
Comment (edquist) [inline rpm/xrootd-lcmaps.spec:55]: Is this redundant with what's in CMakeLists.txt?  Or is `-DOPENSSL111=1` needed in both places?

Comment (efajardo) [reply rpm/xrootd-lcmaps.spec:55]: it is needed in both places

**Comment by edquist** [comment]
Can't vouch for correctness, but looks like it does what was intended

File: src/GlobusSupport.cc [hunk 4]
```diff
@@ -315,7 +315,12 @@ class CertStore {
                                   globus_gsi_callback_create_proxy_callback);
     X509_STORE_set_depth(m_cert_store, GLOBUS_GSI_CALLBACK_VERIFY_DEPTH);
     X509_STORE_set_flags(m_cert_store, X509_V_FLAG_ALLOW_PROXY_CERTS);
-    m_cert_store->check_issued = globus_gsi_callback_check_issued;
+
+    #if OPENSSL_VERSION_NUMBER < 0x1110000
```
Comment (bbockelm) [inline src/GlobusSupport.cc:319]: I think you want this:
```suggestion
    #if OPENSSL_VERSION_NUMBER < 0x010101000
```
if you want 1.1.1 ; `0x1110000` is 0.17.16 if I'm reading the numbers correctly.

Comment (efajardo) [reply src/GlobusSupport.cc:319]: Ahh you are right

**Comment by efajardo** [comment]
@bbockelm I compiled it in centos8 and travis on el7 so we are good.

**Pull request merged**